<a href="https://colab.research.google.com/github/wahyunh10/Medical-Insurance-Cost-Project/blob/main/Medical_Insurance_Cost_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Medical Insurance Cost Project**

# **About The Project**

---



In 2019, the number of people with health insurance in the United State was close to 300 milion, about 92 percent of the population. Across the United States, Americans pay different premiums monthly for health insurance. The cost of medical insurance can very based on certain factors. Some of the factor that accoount for costs are age, location, tobacco use, dependents and plan category.

In this project, we will analyze individual medical insurance cost and variables that may affect it. This project goal are to:

* Analyze multiple variables in meaningful ways that will provide insight into the dataset
* Implement Data Analys skill in a real-world scenario
* Utilize Python to model this data in a way that is digestible

Each section of this notebook will contain code that is being used to analyzw the dataset, as well as any findings. The aims is to provide both insights into the data while also showingcasing coding and data analysis skill.

# **The Dataset**

---



The dataset is obtained from [Kaggle dataset](https://www.kaggle.com/datasets/mirichoi0218/insurance). The dataset consist of 7 columns.

* `age`: age of primary beneficiary
* `sex`: insurance contractor gender, male or female
* `bmi`: body mass index, providing an understandg of body, weights that relatively high or low relative to height, objective index of body weight (kg/m^2) using the ratio of height to weight
* `children`: number of children covered by health insurance/number if dependents
* `smoker`: is the beneficiary smoking ? yes or no
* `region`: the beneficiary's residential area in the US; northeast southeast southwest, northwest
* `charges`: indiidual medical cost billed by health insurance in USD

# **Import The Dataset**

---

Before begin analyzing our dataset, the first thing we have to do  is import the dataset. Then we can see the header and example of row in the dataset. For example, we will take the first row.




In [1]:
import csv

with open('insurance.csv', newline='') as insurance_csv:
    csv_reader = csv.reader(insurance_csv)
    csv_headings = next(csv_reader)
    first_line = next(csv_reader)
    print(csv_headings)
    print(first_line)

['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges']
['19', 'female', '27.9', '0', 'yes', 'southwest', '16884.924']


From example above, we can see the insurance_csv file has 7 columns. We can also decide the data type for each column; `age` and `children` are integers, `bmi` and `charges` are floats, and `sex`, `smoker` and `region` are strings.

We will make separate list for each column.


In [5]:
# create list for each column
age = []
sex = []
bmi = []
children = []
smoker = []
region = []
charges = []

In [7]:
# read the dataset
with open('insurance.csv', newline='') as insurance_csv:
    insurance_dict = csv.DictReader(insurance_csv)
    for data in insurance_dict: # append row from dictionary to list 
        age.append(int(data['age']))
        sex.append(data['sex'])
        bmi.append(float(data['bmi']))
        children.append(int(data['children']))
        smoker.append(data['smoker'])
        region.append(data['region'])
        charges.append(float(data['charges']))

# **Investigating The Dataset**

---

Let's see if the dataset has missing data or not


In [29]:
from typing import List
complete_list_dict = {'age' : age,
                      'sex' : sex,
                      'bmi' : children,
                      'children' : children,
                      'smoker' : smoker,
                      'region' : region,
                      'charges': smoker}

for key in complete_list_dict:
    print("The number of data in {key} list is {list_length}".format(key=key, list_length=len(complete_list_dict[key])))

The number of data in age list is 1338
The number of data in sex list is 1338
The number of data in bmi list is 1338
The number of data in children list is 1338
The number of data in smoker list is 1338
The number of data in region list is 1338
The number of data in charges list is 1338


There are 1338 data in all columns. Since the number is same across all columns, the are no missing data in the dataset

In [30]:
total_population = len(age)
print(total_population)

1338


We have imported the dataset, now we can begin analyzing

# **Statistical Summary**

---

First, we can perform statistical analysis of data in each column.

* For categorical column, we can sum the value in each category.
* For numerical column, we can evaluate its min, max and average value.




# **Sex Distribution**

In [31]:
# check the unique value in sex column
sex_unique = []
for sx in sex:
    if sx not in sex_unique:
        sex_unique.append(sx)
print(sex_unique)

['female', 'male']


In [32]:
# check the distribution of sex
female_num = sex.count('female')
male_num =sex.count('male')
female_pct = round(female_num/total_population*100,2)
male_pct = round(male_num/total_population*100,2)
print("There are {count} female data, which contibutes to {pct}% of total data"
      .format(count=female_num, pct=female_pct))
print("There are {count} male data, which contributes to {pct}% of total data"
      .format(count=male_num,pct=male_pct))

There are 662 female data, which contibutes to 49.48% of total data
There are 676 male data, which contributes to 50.52% of total data


# **Region Distribution**

In [34]:
# check the unique value in region column
region_unique = []
for reg in region:
    if reg not in region_unique:
        region_unique.append(reg)
print(region_unique)

['southwest', 'southeast', 'northwest', 'northeast']


In [35]:
# check the distribution of region
sw_count = region.count('southwest')
se_count = region.count('southeast')
nw_count = region.count('northwest')
ne_count = region.count('northeast')

sw_pct = round(sw_count/total_population*100,2)
se_pct = round(sw_count/total_population*100,2)
nw_pct = round(sw_count/total_population*100,2)
ne_pct = round(sw_count/total_population*100,2)      

print("There are {count} people from Southwest region, or {pct}% of total population"
      .format(count=sw_count, pct=sw_pct))
print("There are {count} people from Southeast region, or {pct}% of total population"
      .format(count=se_count, pct=se_pct))
print("There are {count} people from Northwest region, or {pct}% of total population"
      .format(count=nw_count, pct=nw_pct))
print("There are {count} people from Northeast region, or {pct}% of total population"
      .format(count=ne_count, pct=ne_pct))

There are 325 people from Southwest region, or 24.29% of total population
There are 364 people from Southeast region, or 24.29% of total population
There are 325 people from Northwest region, or 24.29% of total population
There are 324 people from Northeast region, or 24.29% of total population


Like sex, the distribution in region is also quite equal.

# **Smoker Distribution**

In [36]:
# check the unique value i smoker column
smoker_unique = []
for smk in smoker:
    if smk not in smoker_unique:
        smoker_unique.append(smk)
print(smoker_unique)

['yes', 'no']


In [39]:
smoker_yes = smoker.count('yes')
smoker_yes_pct = round(smoker_yes/total_population*100,2)

print("{smoker} peopele are smoker, which is {pct}% of total population"
      .format(smoker=smoker_yes, pct=smoker_yes_pct))

274 peopele are smoker, which is 20.48% of total population


Moving forward to **numerical columns**, let's start with age.

# **Age Distribution**

In [40]:
age_min = min (age)
age_max = max (age)
age_avg = round(sum(age)/total_population,2)
print ("The average in the population is {age} years old".format(age=age_min))
print ("The average in the population is {age} years old".format(age=age_max))
print ("The average in the population is {age} years old".format(age=age_avg))

The average in the population is 18 years old
The average in the population is 64 years old
The average in the population is 39.21 years old
